adapters/repos/db/repo.go

```diff
@@ -76,6 +76,31 @@ type DB struct {
 	ratePerSecond       int
 }
 
+func (db *DB) GetSchemaGetter() schemaUC.SchemaGetter {
+	return db.schemaGetter
+}
+
+func (db *DB) GetSchema() schema.Schema {
+	return db.schemaGetter.GetSchemaSkipAuth()
```

Comment:
Are we certain there is no harm in side-stepping `db.schemaGetter.GetSchemaSkipAuth()` here? I have seen elsewhere in the codebase that getting the schema is a protected action, and skipping auth is only accessible by some privileged packages, which are assumed to have already authenticated/authorized

Reply:
The other examples don't appear to be any more protected than this one, e.g. c9213605e70346188755d4890cc72b0f5cecef2c (ListClasses), is in the same package, and does not appear to have protection.

I also don't see the point of protecting the schema in preference to the data, it's not like we keep passwords in the schema.  

It looks like the API on schemaGetter is not fitting, since we have authentication on the network calls, not on the database itself.

Can you point out the potential vulnerability here?  It's not like users can call arbitrary functions.